rclcpp/include/rclcpp/subscription_base.hpp

```diff
@@ -234,6 +234,18 @@ class SubscriptionBase : public std::enable_shared_from_this<SubscriptionBase>
   void
   handle_loaned_message(void * loaned_message, const rclcpp::MessageInfo & message_info) = 0;
 
+// Take the loaned message
+/*
+ * Similar to take_type_erased, but this is a zero copy method that doesn't rely on the pointers on the stack, and doesn't use messages[out]
```

Comment:
IMO, This depends on middleware, it could be zero copy that that does not promise via interface. we can only say that incoming messages can be loaned from middleware to the user application.